### PR TITLE
Use custom config for publish job

### PR DESCRIPTION
**Pull request by gkhunger-sf**
Dry run of npm publish with these settings worked as expected - https://app.circleci.com/pipelines/github/SalesforceCommerceCloud/commerce-sdk/1282/workflows/05fe13d5-f2d2-433b-8183-356f567bdf17/jobs/2115

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -59,7 +59,10 @@ jobs:
       - test
 
   publish:
-    executor: node-10
+    executor:
+      name: node/default
```
Comment (dbarbato-sf) [inline .circleci/config.yml:63]: Just curious, what does node/default refer to?

File: .circleci/config.yml [hunk 2]
```diff
@@ -59,7 +59,10 @@ jobs:
...
+      tag: '10.20.1'
```
Comment (dbarbato-sf) [inline .circleci/config.yml:64]: Also curious what this is referring to.

File: .circleci/config.yml (hunk 1 above)
Comment (gkhunger-sf) [reply .circleci/config.yml:63]: It's the pre-built docker image we want circle to use

File: .circleci/config.yml (hunk 2 above)
Comment (gkhunger-sf) [reply .circleci/config.yml:64]: circle has image tags for each node version. Here's the complete list of pre-built docker images and tags - https://circleci.com/docs/2.0/docker-image-tags.json. With `name` and `tag`, we are telling circle to use the docker image that has the node version we want to use.

**Pull request merged**